### DISCUSSION TITLE
VS Code Marketplace API

### DISCUSSION THREAD

**Discussion by azat-io** [microsoft/vscode-discussions#875] [Extension Development QnA] [answered]
Hi there!

Is there any way to download VS Code extension using Node.js?

1. Maybe you have VS Code Marketplace API like GitHub API?
2. Or maybe I can download extension from command line without installation? Something like `code --install-extension {publisher}.{name}`

**Comment by tamuratak** [comment]
> 1. Maybe you have VS Code Marketplace API like GitHub API?

Not supported.

> 2. Or maybe I can download extension from command line without installation? Something like code --install-extension {publisher}.{name}

Yes. See https://code.visualstudio.com/docs/editor/extension-marketplace#_command-line-extension-management

> code --install-extension (extension-id | extension-vsix-path)     
   Installs an extension.

**Reply by azat-io** [reply]
I want to download the extension, not install it.

**Reply by r3m0t** [reply, marked as answer]
You can do this with https://open-vsx.org/, but, there is a license on VS Code Marketplace which may be relevant - https://aka.ms/vsmarketplace-ToU

**Reply by isidorn** [reply]
This is against the VS Marketplace terms of use.
